sqlite.go

```diff
@@ -1068,3 +1093,20 @@ func WithQueryCancel(ctx context.Context) context.Context {
 
 // queryCancelKey is a context key for query context enforcement.
 type queryCancelKey struct{}
+
+// ConnLogger is implemented by the caller to support statement-level logging for
+// write transactions. Only Exec calls are logged, not Query calls, as this is
+// intended as a mechanism to replay failed transactions.
+type ConnLogger interface {
```

Comment:
We probably want some kind of Close() method to ensure the file is flushed to disk when we exit.